src/coreclr/jit/gentree.cpp

```diff
@@ -7239,6 +7239,15 @@ bool GenTree::OperMayThrow(Compiler* comp)
         helper = comp->eeGetHelperNum(this->AsCall()->gtCallMethHnd);
         return ((helper == CORINFO_HELP_UNDEF) || !comp->s_helperCallProperties.NoThrow(helper));
     }
+#ifdef FEATURE_HW_INTRINSICS
+    else if (OperIsHWIntrinsic())
+    {
+        if ((gtFlags & (GTF_HW_USER_CALL | GTF_EXCEPT)) == (GTF_HW_USER_CALL | GTF_EXCEPT))
```

Comment:
`OperMayThrow` is expected to recompute whether we need to set `GTF_EXCEPT` on a node, without taking into account its operand nodes. This does take into account the operands nodes; I think instead the check should be:
```suggestion
        if ((gtFlags & GTF_HW_USER_CALL) != 0)
```